airflow-core/src/airflow/models/trigger.py

```diff
@@ -366,9 +386,9 @@ def get_sorted_triggers(cls, capacity: int, alive_triggerer_ids: list[int] | Sel
         )
         asset_triggers = session.execute(query).all()
 
-        # Add triggers associated to assets after triggers associated to tasks
-        # It prioritizes DAGs over event driven scheduling which is fair
-        return ti_triggers + asset_triggers
+        # Add triggers associated to deadlines first, then tasks, then assets
+        # It prioritizes deadline triggers, then DAGs over event driven scheduling which is fair
+        return deadline_triggers + ti_triggers + asset_triggers
```

Comment:
Do we need to limit this list to `capacity` length?  We're limiting the deadlines results, but could then go over capacity when adding the others, right?

Reply:
I thought about that but I noticed that asset triggers were not accounting for this either so I assumed this is probably the correct way. So I'm not sure but looking into it

Reply:
If this is how it is done in other places, I'm willing to be wrong. Feel free to resolve this one but maybe leave it open till the end in case someone else has a thought?